src/auth/scheme/validate_user.ts

```diff
@@ -29,26 +31,36 @@ type ValidateUser = (a: {userId: number, organisationId: number}, b: Hapi.Reques
 
 const validateUser: ValidateUser = async (decoded, request) => {
   try {
-
+    const { knex } = request;
     const { userId, organisationId } = decoded;
-    if (userId && organisationId) {
-
-      const rolesInterface = rolesInitialiser(request.knex);
-      const permissionsInterface = permissionsInitialiser(request.knex);
-      const getRoleId = await rolesInterface.getUserRole({ userId, organisationId });
-      const roleId = getRoleId.access_role_id;
-
-      const getPermissions = await permissionsInterface.permissionsForRole({ roleId });
-      const scope = getPermissions.map(createScopeName);
-      return {
-        credentials: {
-          userId,
-          roleId,
-          scope,
-        },
-        isValid: true };
+
+    if (!userId || !organisationId) {
+      return { isValid: false };
     }
-    return { isValid: false };
+
+    const [
+      user,
+      organisation,
+      userRole,
+    ] = await Promise.all([
+      Users.getOne(knex, { where: { id: userId } }),
+      Organisations.getOne(knex, { where: { id: organisationId } }),
+      Roles.fromUser(knex, { userId, organisationId }),
```

Comment:
The new title of this method (`Roles.fromUser`) reads as though it return an array of associated roles. This is not the case as it grabs the first response. For clarity in calling code I think this could be renamed to something like `Roles.singleRowFromUser`.

Reply:
`singleRow` evokes the database, which is not something we should be concerned about; The module should abstract that away. How's `Roles.oneFromUser`?